src/util/account_util.js

```diff
@@ -475,7 +531,54 @@ function _check_root_account(account) {
     return account.owner === undefined;
 }
 
+function _check_access_key_is_deactivated(status) {
+    return status === ACCESS_KEY_STATUS_ENUM.INACTIVE;
+}
+
+function _get_access_key_status(deactivated) {
+    // we would like the default to be Active (so when it is undefined it would be Active)
+    const status = deactivated ? ACCESS_KEY_STATUS_ENUM.INACTIVE : ACCESS_KEY_STATUS_ENUM.ACTIVE;
+    return status;
+}
+
+function get_filtered_access_keys(requested_account, access_key_id) {
+    const filtered_access_keys = requested_account.access_keys.filter(access_key => access_key.access_key.unwrap() !== access_key_id);
+    if (filtered_access_keys.length > 0) {
+        filtered_access_keys[0].secret_key = system_store.master_key_manager
+            .encrypt_sensitive_string_with_master_key_id(filtered_access_keys[0].secret_key, requested_account.master_key_id._id);
+    }
+    return filtered_access_keys;
+}
 
+function _check_specific_access_key_inactive(action, access_key_to_find, requested_account) {
+    const resource_name = 'access keys';
+    for (const access_key_obj of requested_account.access_keys) {
+        const access_key = access_key_obj.access_key instanceof SensitiveString ?
+                                access_key_obj.access_key.unwrap() : access_key_obj.access_key;
+        if (access_key_to_find === access_key) {
+            if (!_check_access_key_is_deactivated(_get_access_key_status(access_key_obj.deactivated))) {
+                _throw_error_delete_conflict(action, requested_account, resource_name);
+            }
+        }
+    }
+}
+
+function _list_access_keys_from_account(requesting_account, account, on_itself) {
+    const members = [];
+    if (!account.access_keys || account.access_keys.length === 0) {
+        return members;
+    }
+    for (const access_key of account.access_keys) {
+        const member = {
+            username: _returned_username(requesting_account, account.name, on_itself).split(IAM_SPLIT_CHARACTERS)[0],
+            access_key: access_key.access_key instanceof SensitiveString ? access_key.access_key.unwrap() : access_key.access_key,
+            status: _get_access_key_status(access_key.deactivated),
+            create_date: access_key.creation_date ?? account.creation_date,
```

Comment:
You would need to add here a comment why you default this value (in NC it was for the root account manager, but we don't have this case in NB).